### DISCUSSION TITLE
Styling for accordion headers - stack view

### DISCUSSION THREAD

**Discussion by siddharth1903** [vmware/clarity#6337] [Q&A] [answered]
Hi,
Using CDS Core Accordion, i would like to achieve a behaviour where in small screens, we would like to have the whole of the header in one line, while the text inside wraps within the space on offer.

The default behavior wraps the whole of the content next to the icon to the next line...  Have attached an image of the storybook/foundation as actual..Please refer Panel 3

Actual: ![image](https://user-images.githubusercontent.com/26298169/133196922-dd12d4d1-6369-4aab-9b4f-96b5efe4f5e7.png)

Expected: ![image](https://user-images.githubusercontent.com/26298169/133196720-822e86ec-c760-4447-bfe3-6abc9b43f982.png)

From what i have seen, setting flex-wrap to no wrap on the private content inside of the shadow root solves this.. But i would like to know what is the best way to achieve this...

Thanks in advance,
Sid

**Comment by bbogdanov** [comment]
Hi @siddharth1903, 

What you are doing seems like a good solution. 

@coryrylan Do you think it will be a good idea to export a `css prop` for the `flex-wrap` so people don't have to change the `shadow-dom` for future situations like the mentioned above?

**Reply by mathisscott** [reply]
That or we could update the flex-wrap property responsively based on a media query in the CSS...

**Reply by siddharth1903** [reply]
@mathisscott is there a problem applying no wrap on all breakpoints? I dont think it is going to be so much of a problem in cases where there is more width on offer! And even in that case, i believe no wrap could work better for consistency?

**Comment by coryrylan** [comment, marked as answer]
I think the fix here would be to add `wrap:none` to the `cds-layout` on [this line](https://github.com/vmware/clarity/blob/next/packages/core/src/accordion/accordion-header.element.ts#L54). That would allow the projected content to wrap but prevent the content from wrapping past the icon

**Reply by siddharth1903** [reply]
@coryrylan @bbogdanov should i create a defect out of this discussion? or will you guys take it up for an upcoming release! Do let me know what works best

**Reply by bbogdanov** [reply]
#6345

**Reply by bbogdanov** [reply]
Merged!
